src/libsyntax/visit.rs

```diff
@@ -308,11 +308,19 @@ pub fn walk_item<'a, V: Visitor<'a>>(visitor: &mut V, item: &'a Item) {
             visitor.visit_generics(generics);
             visitor.visit_enum_def(enum_definition, generics, item.id, item.span)
         }
-        ItemKind::Impl(_, _, _, ref generics, ref opt_trait_reference, ref typ, ref impl_items) => {
+        ItemKind::Impl {
+            unsafety: _,
+            polarity: _,
+            defaultness: _,
+            ref generics,
+            ref of_trait,
```

Comment:
Not sure what "of" signifies here. I would rename this to `trait_ref` and let the type system speak to the `Option<_>`

Reply:
It's `Some` if the item is an impl of a trait.

Reply:
But what does the f stand for? (I know what the field means.)

Reply:
"of" like the preposition. An impl *of* a trait.

Reply:
Oh. Well that was far from obvious. :) Given that you had to ELI5 it for me, could we change it to something else?

Reply:
I could, although to be honest I'm not enthusiastic about it. On a personal note, I have less time to contribute to open source this year, and having to spend it arguing over trivial choices like this has left me feeling a bit bummed out 😢.

Reply:
(Fwiw, I didn't think we were arguing -- I was just confused as to what it signified.)